lib/parser/parse-tree.h

```diff
@@ -2931,6 +2931,9 @@ struct GenericStmt {
 // R1503 interface-stmt -> INTERFACE [generic-spec] | ABSTRACT INTERFACE
 struct InterfaceStmt {
   UNION_CLASS_BOILERPLATE(InterfaceStmt);
+  // Workaround for clang with libstc++10 bug
+  InterfaceStmt(Abstract x) : u(x) {}
```

Comment:
We typically use curly braces for member initialization in constructors.

The parse tree node types all have move constructors, and intentionally omit copy semantics; does this WAR still succeed if written `InterfaceStmt(Abstract &&x) : u{std::move(x)} {}`?

Reply:
Unfortunately it doesn't as this still triggers the defective behaviour. My first thought was to implement it this way for consistency. However the copy should be cheap/free as Abstract is an empty class.
I can possibly use braces for the member initialisation instead to follow the style guide, I'll check if that still works.